manifests/init.pp

```
@@ -47,25 +47,108 @@
+  Stdlib::Absolutepath $logpath              = '/var/log',
+  String               $user                 = 'puppet',
+  String               $group                = 'puppet',
+  String               $gem_provider         = 'puppet_gem',
+  Optional[String]     $gem_source           = undef,
+  String               $log_level            = 'INFO',
+  Integer              $jwt_token_validity   = 7200,
+  Sensitive[String]     $jwt_token_secret    = Sensitive(fqdn_rand_string(30)),
+  Variant[Sensitive[Hash], Hash] $config     = {},
+) {
+  # install the autosign gem
+  if $autosign::manage_package {
+    package { 'autosign via puppet_gem':
+      ensure   => $autosign::ensure,
+      name     => $autosign::package_name,
+      source   => $autosign::gem_source,
+      provider => $autosign::gem_provider,
+    }
+    package { 'autosign via puppetserver_gem':
+      ensure   => $autosign::puppetserver_ensure,
+      name     => $autosign::package_name,
+      source   => $autosign::gem_source,
+      provider => 'puppetserver_gem',
+    }
+  }
+
+  $dir_ensure = $autosign::ensure ? {
+    /(absent|purged)/ => 'absent',
+    default           => 'directory',
+  }
+
+  $config_ensure = $autosign::ensure ? {
+    /(absent|purged)/ => 'absent',
+    default           => 'file',
+  }
+
+  # Merge the default config with the user-provided config, unwrap before merging 
+  $settings             = Sensitive.new(deep_merge({
+    'general'   => {
+      'loglevel' => $autosign::log_level,
+      'logfile'  => "${autosign::logpath}/autosign.log",
+    },
+    'jwt_token' => {
+      'validity'    => $autosign::jwt_token_validity,
+      'journalfile' => "${autosign::journalpath}/autosign.journal",
+      # THIS IS NOT SECURE! It is marginally better than harcoding a password,
+      # but it can be replicated externaly to the Puppet Master.
+      # Please override this. It will also cause multi-master setups to not work
+      # correctly, all the more reason to override it.
```

This kind of comment should not be needed. The code should instead raise an error to avoid an insecure configuration in the first place.